### PR TITLE
Add automatic deploy on push/pr to main

### LINKED ISSUE #5: Setup automatic deploy on push to main
When changes are pushed to main, they will be compiled, tested on GitHub via GitHub Actions and deployed to Azure.

### PR DESCRIPTION
**Pull request by Hamsterpants**
Did a local merge before i opened this pr, hope everything is good 😊😊😊
closes #5

### CODE REVIEW

File: rest-client-quickstart/src/main/docker/Dockerfile.jvm [hunk 1]
```diff
@@ -81,6 +81,7 @@ FROM registry.access.redhat.com/ubi8/openjdk-21:1.20
 
 ENV LANGUAGE='en_US:en'
 
+ENV HOST 0.0.0.0
```
Comment (muItilingualism) [inline rest-client-quickstart/src/main/docker/Dockerfile.jvm:84]: Is this not the default?

File: rest-client-quickstart/src/main/resources/application.properties [hunk 2]
```diff
@@ -1,8 +1,11 @@
 %prod.quarkus.datasource.db-kind=postgresql
-%prod.quarkus.datasource.username=quarkus
-%prod.quarkus.datasource.password=quarkus
-%prod.quarkus.datasource.jdbc.url=jdbc:postgresql:security_jpaTODO
+%prod.quarkus.datasource.username=postgres
+%prod.quarkus.datasource.password=Yourstrongpassword.
+%prod.quarkus.datasource.jdbc.url=jdbc:postgresql:///postgres?cloudSqlInstance=cs-devops-62582-group8:europe-north1:quarkus&socketFactory=com.google.cloud.sql.postgres.SocketFactory
```
Comment (muItilingualism) [inline rest-client-quickstart/src/main/resources/application.properties:4]: Move to environment variable

**Review by muItilingualism — CHANGES_REQUESTED** [review]
Looks good, minor changes to safekeep IPs and passwords👍

File: rest-client-quickstart/src/main/docker/Dockerfile.jvm (hunk 1 above)
Comment (Hamsterpants) [reply rest-client-quickstart/src/main/docker/Dockerfile.jvm:84]: I think so, it was added just to make sure, i will remove it and hope nothing breaks

Comment (muItilingualism) [reply rest-client-quickstart/src/main/docker/Dockerfile.jvm:84]: If it deploys it should be fine

File: rest-client-quickstart/src/main/resources/application.properties [hunk 3]
```diff
@@ -1,8 +1,11 @@
 %prod.quarkus.datasource.db-kind=postgresql
-%prod.quarkus.datasource.username=quarkus
-%prod.quarkus.datasource.password=quarkus
-%prod.quarkus.datasource.jdbc.url=jdbc:postgresql:security_jpaTODO
+%prod.quarkus.datasource.username=postgres
+%prod.quarkus.datasource.password=Yourstrongpassword.
```
[the remark this answers is not in the document]
Comment (muItilingualism) [reply rest-client-quickstart/src/main/resources/application.properties:3]: Also, after doing this, please change the password, as this repo is public

File: rest-client-quickstart/src/main/resources/application.properties (hunk 2 above)
Comment (muItilingualism) [reply rest-client-quickstart/src/main/resources/application.properties:4]: Just move the"cs-devops-62582-group8:europe-north1:quarkus" part to env var. The rest can stay

Comment (Hamsterpants) [reply rest-client-quickstart/src/main/resources/application.properties:4]: Where should the `.env` file be located, i am a bit lost, did some reading at https://quarkus.io/guides/config-reference#env-file but brain not good.

Comment (muItilingualism) [reply rest-client-quickstart/src/main/resources/application.properties:4]: I was thinking you do the same as with the pem keys. Pass them as flags when launching the gcloud server

Comment (muItilingualism) [reply rest-client-quickstart/src/main/resources/application.properties:4]: Like this

https://github.com/Complex-systems-and-devops-gruppe-8/backend/blob/b663c1e75bd7805af6ccb3526843720fe46f8ecd/.github/workflows/prod.yml#L80

**Review by muItilingualism — APPROVED** [review]

**Pull request merged**